test/unit/multipolicy.cxx

```diff
@@ -0,0 +1,21 @@
+#include "gtest/gtest.h"
+#include "RAJA/RAJA.hxx"
+
+TEST(multipolicy, basic)
+{
+  auto mp = RAJA::make_multi_policy<RAJA::seq_exec,RAJA::omp_parallel_for_exec>(
+      [] (const RAJA::RangeSegment& r) {
```

Comment:
What are your thoughts on a type-aware dispatch pending the `Iterable` type? For example, a list of lambdas which are specialized for `RangeStrideSegment` or `RangeSegment`. It would be similar to the `std::visitor` pattern (observe) found with `std::variant`

```cpp
auto mp = RAJA::make_multi_policy<RAJA::seq_exec, RAJA::omp_parallel_for_exec>(
            RAJA::specialized_multi_policy(
              [] (const RAJA::RangeSegment &r) {
                 //......
              },
              [] (const RAJA::RangeStrideSegment &r) {
                 //......
              }));
```

`specialized_multi_policy` could extract the type arguments of each lambda/callable and invoke the correct one based on the invocation of `forall`